src/mount/writedata.cc

```diff
@@ -66,26 +66,24 @@
 
 namespace {
 
+typedef std::unique_lock<std::mutex> Lock;
+struct ChunkData;
+
 struct inodedata {
 	uint32_t inode;
-	uint64_t maxfleng;
-	int status;
-	uint16_t flushwaiting;
-	uint16_t writewaiting;
-	uint16_t lcnt;
-	uint32_t trycnt;
-	bool inqueue; // true it this inode is waiting in one of the queues or is being processed
-	uint32_t minimumBlocksToWrite;
-	std::list<WriteCacheBlock> dataChain;
-	int alterations_in_chain; // number of adherent blocks with different chunk ids in chain
-	std::condition_variable flushcond; // wait for !inqueue (flush)
-	std::condition_variable writecond; // wait for flushwaiting==0 (write)
-	inodedata *next;
-	std::unique_ptr<WriteChunkLocator> locator;
-	int newDataInChainPipe[2];
-	bool workerWaitingForData;
-	Timer lastWriteToDataChain;
-	Timer lastWriteToChunkservers;
+	uint64_t maxfleng;                  // inodeLock
+	int status;                         // inodeLock
+	uint16_t flushwaiting;              // inodeLock
+	uint16_t writewaiting;              // inodeLock
+	uint16_t lcnt;                      // inodeLock
+	std::condition_variable flushcond;  // wait for !inqueue (flush): using gLock
+	std::condition_variable writecond;  // wait for flushwaiting==0 (write): using inodeLock
+	inodedata *next;  // for iterating through the inodedata hash table, using gLock
+	std::atomic<ChunkData *> chunksDataFirst;
```

Comment:
I'll try to describe a little bit the change I did here:
- change the ```std::atomic<ChunkData *> chunksDataFirst;``` for ```std::list<ChunkDataPtr> chunksDataList```.
- it will allow us to make a lot easier and modern iterations and most of all find operations.
- the ChunkDataPtr is a std::unique_ptr<ChunkData>, meaning that destroying the stored pointer in the list will release the data as wanted. In the other parts of the code will suffice with a copy of the pointer.

I apply a very similar change to this in the idhash variable (```inodedata **idhash``` to ```std::list<InodeDataPtr> idhash[IDHASHSIZE]```).